scripts/java-api-check-all.sh

```diff
@@ -25,8 +32,28 @@ if ! [ -x "$(command -v revapi.sh)" ]; then
     popd > /dev/null
 fi
 
-RELEATE_TAG=${1-""}
-echo "Release tag: $RELEATE_TAG"
+RELEASE_TAG=${1-""}
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+TARGET_BRANCH=$(gh pr view "$CURRENT_BRANCH" --json baseRefName --jq '.baseRefName')
+echo "CURRENT_BRANCH: $CURRENT_BRANCH"
+echo "TARGET_BRANCH: $TARGET_BRANCH"
+
+if ! [ "$TARGET_BRANCH" = "main" ]; then # Release branches
+  LATEST_RELEASE_TAG_IN_THIS_BRANCH=$(git tag --merged "$CURRENT_BRANCH" --sort=-creatordate | grep "android-v"| head -n 1)
```

Comment:
Is the name of the target branch here already sufficient? E.g. `v10.4`

Reply:
No, only with target branch we don't know whether it's a beta/rc or stable release.

Reply:
Ah, so PRs to 10.4.0-beta / 10.4.0-rc we will compare with 10.3, and when 10.4.0 is released and we're working on 10.4.1 - we'll compare with 10.4.0, is that right?

Reply:
That's right.

Reply:
> Ah, so PRs to 10.4.0-beta / 10.4.0-rc we will compare with 10.3, and when 10.4.0 is released and we're working on 10.4.1 - we'll compare with 10.4.0, is that right?

I think it should compare with `10.3` even when we're working on 10.4.1. for example, anyone upgrading from `10.3` to latest stable `10.4.1`, their java code shouldn't break 🤔, any thoughts?

Reply:
The reason why use the last stable release is to exclude any changes(maybe not breaking changes) that could fail the ci job. If 10.3 to 10.4 has the breaking change, it should be found before release 10.4, for  10.4.1 we just need to make sure no breaking change for this patch release.

Reply:
@ank27 suppose in this case we'll have breaking change in 10.3-10.4, right? And if both updates 10.3-10.4 / 10.4-10.4.1 has no breaking changes then it means 10.3-10.4.1 is also safe.
At the same time if there is something breaking between 10.3-10.4 and we decide we allow such change comparing 10.4.1 with 10.3 will force us to review those breaking changes from the past once again

Reply:
correct 👍